recipes/LibriSpeech/SSL/hparams/train_wav2vec.yaml

```diff
@@ -0,0 +1,95 @@
+seed: 1986
+__set_seed: !apply:torch.manual_seed [!ref <seed>]
+data_folder: /idiap/resource/database/LibriSpeech
+save_folder: data
+output_folder: exp-test
+
+train_splits: ["train-clean-100", "train-clean-360", "train-other-500"]
+dev_splits: ["dev-clean"]
+test_splits: ["test-clean", "test-other"]
+train_csv: !ref <output_folder>/train.csv
+valid_csv: !ref <output_folder>/dev-clean.csv
+test_csv:
+   - !ref <output_folder>/test-clean.csv
+   - !ref <output_folder>/test-other.csv
+skip_prep: true
+
+avoid_if_longer_than: 16.
+avoid_if_shorter_than: 2.5
+log_interval: 25
+use_wandb: False
+auto_mix_prec: True
+
+number_of_epochs: 1000
+optimizer_step_limit: 400000
+
+sample_rate: 16000
+lr: 0.0004
+warmup: 20000
+seconds_per_batch: 200
+encoder_layerdrop: 0.05
+
+diversity_loss_weight: 1.
+latent_l2_penalty_weight: 10.
+mask_prob: 0.65
+mask_len: 10
+
+embedding_dim: 768
+final_dim: 256
+latentextractor_kernels: [11, 3, 3, 3, 3, 3, 3]
+latentextractor_strides: [5, 2, 2, 2, 2, 2, 2]
+
+optimizer: !name:torch.optim.AdamW
+  lr: !ref <lr>
+  weight_decay: 0.01
+  eps: 0.000001
+
+test_dataloader_options:
+    batch_size: 8
+    num_workers: 6
+
+epoch_counter: !new:speechbrain.utils.epoch_loop.EpochCounter
+    limit: !ref <number_of_epochs>
+
+extractor: !new:speechbrain.lobes.models.wav2vec.W2VLatentExtractor
+  kernel_sizes: !ref <latentextractor_kernels>
+  strides: !ref <latentextractor_strides>
+
+encoder: !new:speechbrain.lobes.models.wav2vec.W2VLatentEncoder
+  dropout: 0.1
+  layerdrop_prob: !ref <encoder_layerdrop>
+
+quantiser: !new:speechbrain.nnet.quantisers.GumbelVectorQuantizer
+  input_dim: 512
+  num_vars: 320
+  temp_tuple: (2.0, 0.25, 0.999995,)
+  groups: 2
+  vq_dim: !ref <final_dim>
+
+model: !new:speechbrain.lobes.models.wav2vec.Wav2Vec2
+  embedding_dim: !ref <embedding_dim>
+  final_dim: !ref <final_dim>
+  quantiser: !ref <quantiser>
+  latent_extractor: !ref <extractor>
```

Comment:
Do we really need to store the extractor and quantiser in the Wav2vec2 class ? I.e can't we compose the forward explicitly in the compute_forward function instead ? If this is not too messy it's usually better.